Sources/MapboxMaps/Foundation/MapboxMap.swift

```diff
@@ -169,3 +177,47 @@ public final class MapboxMap {
         return rect
     }
 }
+
+// MARK: - ObservableProtocol
+
+extension MapboxMap: ObservableProtocol {
+    public func subscribe(_ observer: Observer, events: [String]) {
+        __map.subscribe(for: observer, events: events)
+    }
+
+    public func unsubscribe(_ observer: Observer, events: [String] = []) {
+        if events.isEmpty {
+            __map.unsubscribe(for: observer)
+        } else {
+            __map.unsubscribe(for: observer, events: events)
+        }
+    }
+}
+
+// MARK: - Map Event handling
+
+extension MapboxMap: MapEventsObservable {
+    @discardableResult
+    public func onNext(_ eventType: MapEvents.EventKind, handler: @escaping (Event) -> Void) -> Cancelable {
+        let handler = MapEventHandler(for: [eventType.rawValue],
+                                      observable: self,
+                                      handler: { event in
+                                        handler(event)
+                                        return true
+                                      })
+        eventHandlers.add(handler)
+        return handler
+    }
+
+    @discardableResult
+    public func onEvery(_ eventType: MapEvents.EventKind, handler: @escaping (Event) -> Void) -> Cancelable {
+        let handler = MapEventHandler(for: [eventType.rawValue],
+                                      observable: self,
+                                      handler: { event in
+                                        handler(event)
+                                        return false
+                                      })
```

Comment:
nit: could we do a trailing closure here?

Reply:
Done.